domains/schema/xWidgetSchema.ts

```diff
@@ -3,7 +3,12 @@ import { z } from 'zod'
 export const xWidgetSchema = z
   .object({
     src: z.string(),
-    type: z.string().optional(),
+    type: z.string(),
```

Comment:
`type` and `username` needs to stay optional, the schema is defined for the actual user input where we only have `src` only.